src/plone/restapi/deserializer/dxcontent.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+from AccessControl import getSecurityManager
+from plone.autoform.interfaces import WRITE_PERMISSIONS_KEY
+from plone.dexterity.interfaces import IDexterityContent
+from plone.dexterity.utils import iterSchemata
+from plone.restapi.deserializer import json_body
+from plone.restapi.interfaces import IDeserializeFromJson
+from plone.restapi.interfaces import IFieldDeserializer
+from plone.supermodel.utils import mergedTaggedValueDict
+from z3c.form.interfaces import IDataManager
+from z3c.form.interfaces import IManagerValidator
+from zExceptions import BadRequest
+from zope.component import adapter
+from zope.component import queryMultiAdapter
+from zope.component import queryUtility
+from zope.event import notify
+from zope.interface import Interface
+from zope.interface import implementer
+from zope.lifecycleevent import ObjectModifiedEvent
+from zope.schema import getFields
+from zope.schema.interfaces import ValidationError
+from zope.security.interfaces import IPermission
+
+
+@implementer(IDeserializeFromJson)
+@adapter(IDexterityContent, Interface)
+class DeserializeFromJson(object):
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+
+        self.sm = getSecurityManager()
+        self.permission_cache = {}
+
+    def __call__(self):
+        data = json_body(self.request)
+
+        modified = False
+        schema_data = {}
+        errors = []
+
+        for schema in iterSchemata(self.context):
+            write_permissions = mergedTaggedValueDict(
+                schema, WRITE_PERMISSIONS_KEY)
+
+            for name, field in getFields(schema).items():
+
+                field_data = schema_data.setdefault(schema, {})
+
+                if field.readonly:
```

Comment:
Right now the behavior is to fail silently if possible, wich is a personal preference of me. This also affects updating an inexistent field or a field you don't have write permission for. I agree that in some cases it may be desirable to have strict error checking. So maybe we should provide an option for strict checking with a `strict=True` URL parameter.

I'd like to discuss error handling separately as it involves other parts than deserialization.
We should also determine first in what format errors should be reported. One of `application/api-problem+json` or `application/vnd.error+json`?